common/tier4_autoware_utils/src/math/trigonometry.cpp

```diff
@@ -49,4 +49,34 @@ float cos(float radian)
   return sin(radian + static_cast<float>(tier4_autoware_utils::pi) / 2.f);
 }
 
+std::pair<float, float> sin_and_cos(float radian)
+{
+  constexpr float tmp =
+    (180.f / static_cast<float>(tier4_autoware_utils::pi)) * (discrete_arcs_num_360 / 360.f);
+  const float degree = radian * tmp;
+  size_t idx =
+    (static_cast<int>(std::round(degree)) % discrete_arcs_num_360 + discrete_arcs_num_360) %
+    discrete_arcs_num_360;
+
+  float sin, cos;
+  if (idx < discrete_arcs_num_90) {
+    sin = g_sin_table[idx];
+    cos = g_sin_table[discrete_arcs_num_90 - idx];
```

Comment:
Can you
```
return {g_sin_table[idx], <cos>}
```
in each block and remove uninitialised `float sin, cos` ?

Reply:
Sorry for my late response.
I have made the changes :+1: